src/serializer.js

```diff
@@ -255,6 +261,29 @@ export class Serializer {
   _notNameChars =
     (this._notQNameChars + ':')
 
+  // Validate if a string is a valid PN_LOCAL per Turtle 1.1 spec
+  // Allows dots inside the local name but not as trailing character
+  // Also allows empty local names (for URIs ending in / or #)
+  isValidPNLocal(local) {
+    // Empty local name is valid (e.g., ex: for http://example.com/)
```

Comment:
Note that resolution of this "empty" local name is determined server-side. It often defaults to `index.html`, but not always. This is configurable in Apache and some (most? all?) other HTTP servers, so it cannot be relied upon without out-of-band communications. Configured local name can be things like `index.php`.